src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,28 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> getResponseConnect(ConnectionInfo connectionInfo, ConnectParams connectionParams)

```

Comment:
I'd recommend renaming method 'getResponseConnect' to "TryOpenConnectionWithRetry" and 'firstRun" to "isRetriable".